stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java

```diff
@@ -0,0 +1,177 @@
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/**
+ * Native function ballerina.compression:unzipBytes.
+ *
+ * @since 0.970.0
+ */
+@BallerinaFunction(
+        orgName = "ballerina", packageName = "compression",
+        functionName = "unzipBytes",
+        args = {@Argument(name = "content", type = TypeKind.BLOB),
+                @Argument(name = "destDir", type = TypeKind.STRING),
+                @Argument(name = "folderToUnzip", type = TypeKind.STRING)},
+        isPublic = true
+)
+public class UnzipBytes extends BlockingNativeCallableUnit {
+    /**
+     * File content as byte array defined.
+     */
+    private static final int SRC_AS_BYTEARRAY_FIELD_INDEX = 0;
+    /**
+     * File path of the destination directory.
+     */
+    private static final int DEST_PATH_FIELD_INDEX = 0;
+
+    /**
+     * Folder to unzip from the compressed bytes.
+     */
+    private static final int FOLDER_TO_UNZIP_INDEX = 1;
+
+    /**
+     * Decompress/unzip byte arrays/blob.
+     *  @param fileContentAsByteArray file content as a byte arry
+     * @param outputFolder           destination folder
+     * @param folderToUnzip folder to unzip in the zipped file
+     */
+    static void decompress(byte[] fileContentAsByteArray, String outputFolder, String folderToUnzip) {
+        ZipInputStream zin = null;
+        try {
+            Path outdir = Paths.get(outputFolder);
+            zin = new ZipInputStream(new ByteArrayInputStream(fileContentAsByteArray));
+            ZipEntry entry;
+            String name, dir;
+            while ((entry = zin.getNextEntry()) != null) {
+                name = entry.getName();
+                if (!folderToUnzip.isEmpty() && name.startsWith(folderToUnzip)) {
+                    int index = name.lastIndexOf('/') + 1;
+                    name = name.substring(index);
+                }
+                    if (entry.isDirectory()) {
+                        mkdirs(outdir, name);
+                        continue;
+                    }
+                    dir = getDirectoryPath(name);
+                    if (dir != null) {
+                        mkdirs(outdir, dir);
+                    }
+                    extractFile(zin, outdir, name);
+            }
+        } catch (IOException e) {
+            throw new BLangRuntimeException("I/O Exception when processing files");
+        } finally {
+            try {
+                if (zin != null) {
+                    zin.close();
+                }
+            } catch (IOException e) {
+                throw new BLangRuntimeException("I/O Exception when closing the input stream " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Extract files from the zipInputStream.
+     *
+     * @param in     zipInputStream object
+     * @param outdir output directory file
+     * @param name   name of the file
+     */
+    private static void extractFile(ZipInputStream in, Path outdir, String name) {
```

Comment:
See if you can use https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#copy(java.io.InputStream,%20java.nio.file.Path,%20java.nio.file.CopyOption...) instead

Reply:
Fixed with bc105af